test/jdk/java/lang/instrument/RetransformGenericSignatureTest.java

```diff
@@ -0,0 +1,233 @@
+/*
```

Comment:
Reworked the test:
- used in-memory compirer and asm to prepare new version of the class;
- used redefineClass instead of retransformClasses (ClassFileTransformer is not needed anymore);
- used RedefineClassHelper agent instead of the agent from ATransformerManagementTestCase framework;
- moved the test to test/hotspot/jtreg/serviceability/jvmti/RedefineClasses as actually it tests JVMTI functionality.